### PR TITLE
[Chat][feature] Participants list

### PR DESCRIPTION
**Pull request by dhiraj**
## Purpose

* Displays participants list on click of participants count and returns back to messages list on tapping again on the count button or the back button

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull Request Checklist

- [ ] Public API changes
- [ ] Verified for cross-platform
- [ ] Synced with iOS, Web team
- [ ] Internal review completed
- [ ] UI Changes
  - [ ] Screen captures included
  - [ ] Tested for Light/Dark mode
  - [ ] Tested for screen rotation
  - [ ] Tested on Tablet and small screen device (5")
  - [ ] Include localization changes
- [ ] Tests
  - [ ] Memory leak analysis performed
  - [ ] Tested on API 21, 26 and latest 
  - [ ] Tested for background mode
  - [ ] Unit Tests Included
  - [ ] UI Automated Tests Included

## How to Test

* Join a chat thread that has participants
* Tap on the participants count to show/hide the participants

## What to Check
Verify that the following are valid
* All the participants are showing as per count
* Appear / Dismiss of participants list works as expected
* Visual layout and alignment of composable works as expected

### CODE REVIEW

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/presentation/ui/chat/components/ParticipantsListView.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.tooling.preview.Preview
+import com.azure.android.communication.ui.chat.models.RemoteParticipantInfoModel
+import com.azure.android.communication.ui.chat.service.sdk.wrapper.CommunicationIdentifier
+
+@Composable
+internal fun ParticipantsListView(
```
Comment (iaulakh) [inline azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/presentation/ui/chat/components/ParticipantsListView.kt:16]: @dhiraj can we use Fluent UI list controls?

**Review by iaulakh — CHANGES_REQUESTED** [review]
Detected crash on screen rotate:
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.azure.android.communication.ui.callingcompositedemoapp.debug, PID: 17358
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.azure.android.communication.ui.callingcompositedemoapp.debug/com.azure.android.communication.ui.chat.presentation.ChatCompositeActivity}: java.lang.IllegalArgumentException: Builder for class com.azure.android.communication.ui.chat.ChatComposite Does not exist
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4035)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
        at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:6347)
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:6212)
        at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:71)
        at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:226)
        at android.os.Looper.loop(Looper.java:313)
        at android.app.ActivityThread.main(ActivityThread.java:8663)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
     Caused by: java.lang.IllegalArgumentException: Builder for class com.azure.android.communication.ui.chat.ChatComposite Does not exist
        at com.azure.android.communication.ui.chat.presentation.ChatCompositeActivity$chatComposite$2.invoke(ChatCompositeActivity.kt:40)
        at com.azure.android.communication.ui.chat.presentation.ChatCompositeActivity$chatComposite$2.invoke(ChatCompositeActivity.kt:15)
        at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
        at com.azure.android.communication.ui.chat.presentation.ChatCompositeActivity.getChatComposite(ChatCompositeActivity.kt:15)
        at com.azure.android.communication.ui.chat.presentation.ChatCompositeActivity.onCreate(ChatCompositeActivity.kt:22)
        at android.app.Activity.performCreate(Activity.java:8290)
        at android.app.Activity.performCreate(Activity.java:8270)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4009)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201) 
        at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:6347) 
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:6212) 
        at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:71) 
        at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loopOnce(Looper.java:226) 
        at android.os.Looper.loop(Looper.java:313) 
        at android.app.ActivityThread.main(ActivityThread.java:8663) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135) 
I/Process: Sending signal. PID: 17358 SIG: 9

**Comment by iaulakh** [comment]
@dhiraj Fima shows people as header when list opens:
Screen Shot 2022-10-24 at 12 58 19 PM

**Review by ahammer — DISMISSED** [review]

**Comment by dhiraj** [comment]
Now using FluentUI controls to render the participants list. I've also updated it to match FIgma header. Further the ActionBar has been refactored to now simply accept title and subTitle and ChatScreen creates and sends the appropriate strings. Screenshots:

Screenshot_1666740074

Screenshot_1666740077

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/state/ParticipantsState.kt [hunk 2]
```diff
@@ -7,5 +7,6 @@ import com.azure.android.communication.ui.chat.models.RemoteParticipantInfoModel
 
 internal data class ParticipantsState(
     val participants: Map<String, RemoteParticipantInfoModel>,
-    val participantTyping: Set<String>
+    val participantTyping: Set<String>,
+    val participantsListVisible: Boolean,
```
Comment (JoshuaLai) [inline azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/state/ParticipantsState.kt:11]: Should the ParticipantState hold onto a value for whether or not the list is visible? 
This seems incorrect to me. Whether or not the participant view is visible is a view concept and has nothing to do with the Participants state. @ahammer  @iaulakh  thoughts?

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/presentation/ui/chat/components/ActionBarView.kt [hunk 3]
```diff
@@ -47,23 +54,25 @@ internal fun ActionBarView(
...
             title = {
                 Column(
-                    modifier = Modifier.padding(start = 90.dp),
+                    modifier = Modifier
+                        .padding(start = 90.dp)
+                        .clickable {
+                            if (!isShowingParticipants) {
+                                postAction(
+                                    ParticipantAction.ParticipantsListVisibilityChanged(visible = true)
+                                )
```
Comment (JoshuaLai) [inline azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/presentation/ui/chat/components/ActionBarView.kt:63]: Would it be more correct to dispatch a navigation action to go to the participant list instead?

**Review by ahammer — DISMISSED** [review]

**Review by JoshuaLai — APPROVED** [review]

**Review by iaulakh — APPROVED** [review]

**Pull request merged**